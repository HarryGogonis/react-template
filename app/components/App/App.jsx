import React from 'react';

const propTypes = {
  children: React.PropTypes.element.isRequired,
};

const App = ({ children }) => children;

App.propTypes = propTypes;

export default App;
