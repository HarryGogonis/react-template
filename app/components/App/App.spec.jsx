import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

let wrapper;
const Component = (
  <App>
    <h1>Child of App</h1>
  </App>
);

describe('<App />', () => {
  beforeEach(() => {
    wrapper = shallow(Component);
  });

  it('should be a function', () => {
    expect(typeof wrapper.type()).toBe('function');
  });

  it('should display the child node', () => {
    expect(wrapper.children.length).toBe(1);
  });

  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(Component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
