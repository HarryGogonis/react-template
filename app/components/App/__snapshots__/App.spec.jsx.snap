// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
<h1>
  Child of App
</h1>
`;
